src/main/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateFlight.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.service.dataset.flight.update;
+
+import bio.terra.model.DatasetSchemaUpdateModel;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.DatasetTableDao;
+import bio.terra.service.dataset.flight.LockDatasetStep;
+import bio.terra.service.dataset.flight.UnlockDatasetStep;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import java.util.UUID;
+import org.springframework.context.ApplicationContext;
+
+public class DatasetSchemaUpdateFlight extends Flight {
+
+  public DatasetSchemaUpdateFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+
+    ApplicationContext appContext = (ApplicationContext) applicationContext;
+    DatasetDao datasetDao = appContext.getBean(DatasetDao.class);
+    DatasetTableDao datasetTableDao = appContext.getBean(DatasetTableDao.class);
+    DatasetService datasetService = appContext.getBean(DatasetService.class);
+    BigQueryDatasetPdao bigQueryDatasetPdao = appContext.getBean(BigQueryDatasetPdao.class);
+
+    DatasetSchemaUpdateModel updateModel =
+        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), DatasetSchemaUpdateModel.class);
+
+    UUID datasetId =
+        UUID.fromString(inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class));
+
+    addStep(new DatasetSchemaUpdateValidateModelStep(datasetService, datasetId, updateModel));
```

Comment:
What happens if two flights tried to add the same table? Should we do this check after locking the dataset?

Reply:
Its an exclusive lock, so the first one will lock, and the second one will fail.

Reply:
Hmm it looks like the dataset lock exception is a retry-able failure whether or not it is an exclusive lock: https://github.com/DataBiosphere/jade-data-repo/blob/2e69cad934e7748c9c7b7262da952914e46abc86/src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java#L63

Reply:
By calling the step in the flight without a retry strategy, it won't retry at all. 

https://github.com/DataBiosphere/stairway/blob/develop/stairway/src/main/java/bio/terra/stairway/Flight.java#L34-L37

Reply:
I agree with Saman...I think logically, it makes sense to lock early for a flight like this and do the validation inside of the scope of the lock (both calls could be making the same update and doing validation outside of the lock I don't think would catch the issue)

Reply:
Fair. I was trying to lock the dataset as late as possible, since the exclusive lock is pretty disruptive. I guess the validation will take like 0.5ms, so it doesn't buy much.

Reply:
Yeah, but a DDL operation _should_ be disruptive